### PR TITLE
release notes: Mention the Nix update in 23.05

### PR DESCRIPTION
**Pull request by thufschmitt**
Links to the corresponding release notes, and mentions https://github.com/NixOS/nix/issues/8259 which isn't included in it (and is somewhat important since it's a language semantics change)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by RaitoBezarius — APPROVED** [review]
Thank you!

**Comment by fricklerhandwerk** [comment]
Fixed conflicts and re-ordered the notes such that those core components appear first.

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 1]
```diff
@@ -12,13 +12,16 @@ To upgrade to the latest release, follow the [upgrade chapter](https://nixos.org
 
 In addition to numerous new and updated packages, this release has the following highlights:
 
+- default Nix: 2.11 -\> 2.13. This includes in particular a [small language change](https://github.com/NixOS/nix/issues/8259) altering the way floats are represented in `builtins.toJSON`.
+  See the release notes for [2.13](https://nixos.org/manual/nix/stable/release-notes/rl-2.13.html) and [2.14](https://nixos.org/manual/nix/stable/release-notes/rl-2.14.html) for more info.
```
Comment (mweinelt) [inline nixos/doc/manual/release-notes/rl-2305.section.md:16]: The 2.14 release notes are 404.

**Comment by mweinelt** [comment]
Included in #235183. Thanks for the reminder.

File: nixos/doc/manual/release-notes/rl-2305.section.md (hunk 1 above)
Comment (mweinelt) [reply nixos/doc/manual/release-notes/rl-2305.section.md:16]: We went with the unstable manual, but please remember to also update stable when you can.

**Pull request closed without merging**